sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +762,72 @@
             return monoError(logger, ex);
         }
     }
+
+    /**
+     * Long Running Operation for the purchase Search.
+     *
+     * @param searchId ID of the search
+     * @param pollInterval Time lapse of the poll request.
+     * @return A {@link PollerFlux} object with the search result
+     */
+
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<Void, Void> beginPurchaseSearch(String searchId, Duration pollInterval) {
+        Objects.requireNonNull(searchId, "'searchId'cannotbenull.");
+        Objects.requireNonNull(pollInterval, "'pollInterval'cannotbenull.");
+
+        return new PollerFlux<Void, Void>(pollInterval,
+            purchaseSearchActivationOperation(searchId),
+            purchaseSearchCreatePollOperation(searchId),
+            (activationResponse, pollingContext) -> Mono.error(new RuntimeException("Cancellationisnotsupported")),
+            purchaseSearchFetchResultOperation(searchId));
+    }
+
+    private Function<PollingContext<Void>, Mono<Void>> purchaseSearchActivationOperation(String searchId) {
+
+        return (pollingContext) -> {
+            Mono<Void> response = purchaseSearchWithResponse(searchId).flatMap(Response -> {
+                Mono<PhoneNumberSearch> phoneNumberSearch = getSearchById(searchId);
+                return null;
+            });
+
+            return response;
+        };
+    }
+
+    private Function<PollingContext<Void>, Mono<PollResponse<Void>>>
+        purchaseSearchCreatePollOperation(String searchId) {
+        return (pollingContext) -> getSearchById(searchId)
+            .flatMap(getSearchResponse -> {
+                LongRunningOperationStatus status = null;
+                SearchStatus statusResponse = getSearchResponse.getStatus();
+                if (statusResponse.equals(SearchStatus.RESERVED)) {
```

Comment:
`RESERVED`, `EXPIRED`, and `SUCCESS ` have the same result. Wouldn't it be better to add all these in a single if statements with an or? Like:  

```
if (statusResponse.equals(SearchStatus.RESERVED) || statusResponse.equals(SearchStatus.EXPIRED) || statusResponse.equals(SearchStatus.SUCCESS)) {
      status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
}
```